open_instruct/vllm_utils_workerwrap.py

```diff
@@ -46,8 +46,10 @@ def init_process_group(
     def update_weight(self, name, dtype, shape, empty_cache=False):
         import torch
 
-        assert dtype == self.model_config.dtype, f"mismatch dtype: src {dtype}, dst {self.model_config.dtype}"
-        weight = torch.empty(shape, dtype=dtype, device="cuda")
+        assert dtype == str(self.model_config.dtype), (
```

Comment:
As discussed earlier, this is done because of weird serialization format in vLLM's async engine.